tools/custom-checks/bump-openapi-spec-dep-version.ts

```diff
@@ -0,0 +1,161 @@
+  const latestVersion = execPromise(
+    `git describe --tags --abbrev=0 ${branchName}`,
+  ).then(({ stdout }) => {
+    const latestTag = stdout.trim().substring(1);
+    console.log(
+      `${TAG} Hyperledger Cacti current version of ${branchName} branch  ${repoUrl} is version ${latestTag}`,
+    );
+    return latestTag;
+  });
+
+  const currentLatestVersion = await latestVersion;
+  const oasPaths = await globby(DEFAULT_GLOB, globbyOpts);
+  console.log(
+    `${TAG} Openapi.json file paths detected (${oasPaths.length}) files`,
+  );
+  console.log(`${TAG} Expected Latest Version: ${currentLatestVersion} \n`);
+  console.log(`${TAG} ---- SCANNING FOR OPENAPI.JSON ---- \n`);
+  console.log(`${TAG} File paths found: `);
+  const errors: string[] = [];
+  const checks = pkgJsonPaths.map(async (pathRel) => {
+    const filePathAbs = path.join(PROJECT_DIR, pathRel);
+    console.log(`${TAG} ${filePathAbs}\n`);
+    const pkgJson: packageJson = await fs.readJSON(filePathAbs);
+    if (typeof pkgJson !== "object") {
+      errors.push(`ERROR: ${pathRel} openapi.json cannot be empty.`);
+      return;
+    }
+    if (!pkgJson) {
+      errors.push(`ERROR: ${pathRel} openapi.json cannot be empty.`);
+      return;
+    }
+    if (!isStdLibRecord(pkgJson)) {
+      return;
+    }
+    const packageVersion = pkgJson.info.version;
+    if (typeof packageVersion == "undefined") {
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} VERSION CHECK FAILED --------: ${pathRel} the package itself has no "version" field. \n`,
+      );
+    } else if (packageVersion == currentLatestVersion) {
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} VERSION CHECK PASSED --------: ${pathRel} the package itself correct has version { ${packageVersion} }. \n`,
+      );
+    } else if (packageVersion != currentLatestVersion) {
+      pkgJson.info.version = currentLatestVersion;
+      await fs.writeFile(filePathAbs, JSON.stringify(pkgJson, null, 2));
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} OUTDATED VERSION DETECTED IN ${pathRel} version: { ${packageVersion} }. Expected: {${currentLatestVersion}} \n`,
+      );
+      console.log(
+        `${TAG} VERSION IS NOW UPDATED --------: ${pathRel} The package now has the latest version { ${pkgJson.info.version} }. \n`,
+      );
+    } else {
+      console.log(`${TAG} ---- CHECKING OPENAPI.JSON VERSION ---- \n`);
+      console.log(
+        `${TAG} VERSION CHECK FAILED --------: ${pathRel} the package itself has no "version" field. Expected version: ${currentLatestVersion} \n`,
+      );
+    }
+    console.log("");
+    const spec = await fs.readJSON(filePathAbs);
+    console.log("");
+    console.log(
+      `${TAG} = = = = CHECKING TOTAL NUMBER OF REFERENCES = = = = \n`,
+    );
+    function countReferences(obj: any): number {
```

Comment:
@aldousalvarez Pleae don't define nested functions like this and instead aim to have a flat structure for them. This make make them have to have more parameters but this is a good thing in the sense that it will make the functions be closer to pure functions.